### PR TITLE
WFLY-18197 + WFLY-18198 Upgrade to Hibernate Search 6.2.0

### PR DESCRIPTION
**Pull request by yrodiere**
* [WFLY-18198](https://issues.redhat.com/browse/WFLY-18198): Upgrade to Lucene 8.11.2
* [WFLY-18197](https://issues.redhat.com/browse/WFLY-18197): Upgrade to Hibernate Search 6.2.0

There is a bit more code than usual for a minor upgrade. That's to handle the passing of Jandex indexes that WildFly builds to Hibernate Search, because Hibernate Search cannot handle the `vfs://` URLs of the WildFly classloaders and would otherwise fail on startup.

That's not so bad because it means users don't need to build Jandex indexes themselves (e.g. with a Maven plugin) and can benefit from WildFly's own Jandex indexes.

See `WildFlyHibernateOrmSearchMappingConfigurer`.

Hey @scottmarlow, could you please tell me if the new `jipijapa-hibernatesearch` module complies with your expectations?

### CODE REVIEW

File: ee-feature-pack/galleon-shared/src/main/resources/modules/system/layers/base/org/hibernate/search/jipijapa-hibernatesearch/main/module.xml
Comment (scottmarlow) [inline ee-feature-pack/galleon-shared/src/main/resources/modules/system/layers/base/org/hibernate/search/jipijapa-hibernatesearch/main/module.xml:5]: should be 2023

File: jpa/hibernatesearch/pom.xml
Comment (scottmarlow) [inline jpa/hibernatesearch/pom.xml:5]: should be 2023

File: jpa/hibernatesearch/src/main/java/org/jipijapa/hibernate/search/HibernateSearchIntegratorAdaptor.java
Comment (scottmarlow) [inline jpa/hibernatesearch/src/main/java/org/jipijapa/hibernate/search/HibernateSearchIntegratorAdaptor.java:3]: should be 2023

File: jpa/hibernatesearch/src/main/java/org/jipijapa/hibernate/search/WildFlyHibernateOrmSearchMappingConfigurer.java
Comment (scottmarlow) [inline jpa/hibernatesearch/src/main/java/org/jipijapa/hibernate/search/WildFlyHibernateOrmSearchMappingConfigurer.java:21]: WildFly only has access to the Jandex indexes at application deployment time.  After the application deploys, Hibernate Search should not have any references to the Jandex indexes since they can be quite large.

File: jpa/spi/src/main/java/org/jipijapa/plugin/spi/PersistenceProviderIntegratorAdaptor.java
Comment (scottmarlow) [inline jpa/spi/src/main/java/org/jipijapa/plugin/spi/PersistenceProviderIntegratorAdaptor.java:3]: Please update Copyright dates to 2023 for changed or new files.

File: jpa/spi/src/main/java/org/jipijapa/plugin/spi/PersistenceProviderIntegratorAdaptor.java
Comment (scottmarlow) [inline jpa/spi/src/main/java/org/jipijapa/plugin/spi/PersistenceProviderIntegratorAdaptor.java:38]: WildFly only has access to the Jandex indexes at application deployment time.  After the application deploys, Hibernate Search should not have any references to the Jandex indexes since they can be quite large.

File: pom.xml
Comment (bstansberry) [inline pom.xml:501]: Note for other reviewers: the .Final version is expected by July 7.

File: ee-feature-pack/galleon-shared/src/main/resources/modules/system/layers/base/org/hibernate/search/mapper/orm/main/module.xml
Comment (bstansberry) [inline ee-feature-pack/galleon-shared/src/main/resources/modules/system/layers/base/org/hibernate/search/mapper/orm/main/module.xml:42]: This comment is just a note for future reference about provisioning. tl;dr; it's all good.

In a slimmed server the org.hibernate.search.mapper.orm module that's being modified here is only provisioned if the hibernate-search layer is provisioned. And this new org.hibernate.search.jipijapa-hibernatesearch is only provisioned because of this dependency. So this new module is handled consistently with the others; it only gets provisioned in an unslimmed server or if the layer is used.

**Comment by yrodiere** [comment]
Thanks for the reviews. I adjusted the copyright notices and made sure references to Jandex indexes are cleaned up after the entity manager factory gets built.
About the failing cloud tests, how can I get information about what went wrong exactly (logs, ...)?

**Comment by scottmarlow** [comment]
I see some changes have Apache License and some have LGPL, was that intentional?

Thanks for ensuring the Jandex indexes will not be held after app deployment completes!

**Comment by yrodiere** [comment]
> I see some changes have Apache License and some have LGPL, was that intentional?

It wasn't. Looks like fixing a PR over the weekend with kids screaming in the next room wasn't the best idea :)
Will update that.

**Comment by yrodiere** [comment]
Here, I changed the licenses of new files to use LGPL everywhere, which is consistent with the Jipijapa SPI module.

**Comment by yrodiere** [comment]
Alright, this is ready for merge @bstansberry :)

If we merge this, once 6.2.0.Final is out I'll prepare an upgrade PR that also makes `hibernate.search.backend.schema.id.strategy` default to `lucene9` (see https://hibernate.atlassian.net/browse/HSEARCH-4884).

**Comment by yrodiere** [comment]
Hey @bstansberry , gentle reminder that you wanted this merged before Wednesday in order for the Search 6.2 upgrade to be in WildFly 29 Beta1.

As far as I can tell Scott doesn't have push access, so we need your or someone else's approval to get this merged.

**Review by bstansberry — APPROVED** [review]
Thanks @yrodiere. LGTM. It was a long weekend in the US, but @darranl was tracking this. When I looked at this before it all seemed fine except the Jandex index caching concern @scottmarlow raised, and you've addressed that. Thanks to both you and Scott!

**Review by scottmarlow — APPROVED** [review]

**Pull request merged**
